linker/jvm/src/main/scala/org/scalajs/linker/irio/SharedZipFile.scala

```diff
@@ -0,0 +1,34 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js linker            **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2018, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.linker.irio
+
+import java.io._
+import java.util.zip.ZipFile
+
+private[irio] final class SharedZipFile(file: File) {
+  private[this] var _references: Int = 0
+  private[this] var _zipFile: ZipFile = _
+
+  final def acquire(): ZipFile = synchronized {
```

Comment:
Have you considered making `acquire()` and `release()` private, and instead expose an API like:
```scala
final def acquireWithin[A](f: ZipFile => A): A = {
  val zipFile = acquire()
  try f(zipFile)
  finally release()
}
```
?

Reply:
Yes. This will not be useful anymore once we move to a fully async world (`release()` would happen in a `andThen` closure). If you want I can change it then.